src/daemon/ca/publishing.rs

```diff
@@ -1002,7 +1010,10 @@ impl BasicKeyObjectSet {
             self.signing_cert = cert.clone();
             Ok(())
         } else {
-            Err(Error::publishing("received new cert for unknown keyid"))
+            Err(Error::PublishingObjects(format!(
+                "received new cert for unknown keyid: {}",
```

Comment:
Elsewhere you construct messages containing "key id" rather than "keyid". Change this to "key id" for consistency?